cmd/observe/observe.go

```diff
@@ -280,12 +292,20 @@ func handleArgs(ofilter *observeFilter) (err error) {
 		if follow {
 			return fmt.Errorf("table output format is not compatible with follow mode")
 		}
-		opts = append(opts, hubprinter.Tab())
+		if nodeName {
+			opts = append(opts, hubprinter.Tab(), hubprinter.WithNodeName())
+		} else {
+			opts = append(opts, hubprinter.Tab())
+		}
 	case "":
 		// no format specified, choose most appropriate format based on
 		// user provided flags
 		if follow {
-			opts = append(opts, hubprinter.Compact())
+			if nodeName {
+				opts = append(opts, hubprinter.Compact(), hubprinter.WithNodeName())
+			} else {
+				opts = append(opts, hubprinter.Compact())
+			}
```

Comment:
I think the option could be added only once instead of individually for every output type, right? (assuming it does not matter to set it for the `json` output)
```go
if nodeName {
    opts = append(opts, hubprinter.WithNodeName())
} 
```